src/utils.js

```diff
@@ -343,3 +348,42 @@ const sleep = (millis) => {
 export const publicUtils = {
     sleep,
 };
+
+/**
+ * Returns a promise that resolves to an array of urls parsed from the resource
+ * available at the provided url.
+ *
+ * @param {String} url
+ * @param {String} encoding
+ * @param {Number} offset
+ * @param {Number} limit
+ * @param {Boolean} skipDuplicates
+ * @param {RegExp} urlRegExp
+ * @returns {Promise<Array>}
+ */
+export const downloadListOfUrls = ({ url, encoding, offset, limit, skipDuplicates, urlRegExp }) => {
+    if (!url) return Promise.reject(new Error('Resource URL must be provided.'));
+    encoding = encoding || 'utf8';
+    offset = offset || 0;
+    return requestPromise.get({ url, encoding })
+        .then((string) => {
+            const urlsArr = extractUrls({ string, urlRegExp }); // not using keep commas for now
+            const limited = urlsArr.slice(offset, limit || undefined);
+            if (!skipDuplicates) return limited;
+
+            const dupeSet = new Set();
+            return limited.filter(u => (dupeSet.has(u) ? false : dupeSet.add(u)));
+        });
+};
+
+/**
+ * Collects all URLs in an arbitrary string to an array.
+ * @param string
+ * @param urlRegExp
+ * @param keepCommas
```

Comment:
I would probably skip `keepCommas` parameter. Because this we may end up having:

`keepCommas`, `keepSpaces`, ... parameters whose only function is to select one of the Regexes

Reply:
I agree with Marek, we can just export the Regexps in utils and that's it.

Reply:
Yes, I agree too. Will make the changes.